tests/perf-tools/geopm/geopm-prun

```diff
@@ -0,0 +1,273 @@
+#!/bin/bash
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# Forked from : components/admin/prun
+# Command line options to mpiexec et. al. were added here for GEOPM as
+# processes must be affinitized to distinct cores for proper functionality.
+# This is normally enforced by the GEOPM launcher.
+
+CONFIG=/etc/sysconfig/prun
+MODE=native
+LOGLEVEL=2     # 0=error,1=warn,2=info,3=debug
+
+function usage () {
+
+    echo " "
+    echo "This OpenHPC utility is used to launch parallel (MPI) applications"
+    echo "within a supported resource manager."
+    echo " "
+    echo "Usage: $DELIM <options> executable [arguments]"
+    echo " "
+    echo "where available options are as follows:"
+    echo " "
+    echo "  -h        generate help message and exit"
+    echo "  -v        enable verbose output"
+    echo " "
+    exit 1
+}
+
+function parse_config_file () {
+
+    if [ -e $CONFIG ];then
+        source $CONFIG 2> /dev/null
+    fi
+}
+
+function _error () {
+    if [ $LOGLEVEL -ge 0 ];then
+        echo -n "[$DELIM] Error: "
+        echo "$@"
+    fi
+}
+
+function _warn () {
+    if [ $LOGLEVEL -ge 1 ];then
+        echo -n "[$DELIM] "
+        echo "$@"
+    fi
+}
+
+function _info () {
+    if [ $LOGLEVEL -ge 2 ];then
+        echo -n "[$DELIM] "
+        echo "$@"
+    fi
+}
+
+function _debug () {
+    if [ $LOGLEVEL -ge 3 ];then
+        echo -n "[$DELIM] "
+        echo "$@"
+    fi
+}
+
+function verify_launcher_avail () {
+    if [ ! $(type -P "$1") ];then
+        _error "Expected Job launcher $1 not found for $LMOD_FAMILY_MPI"
+        exit 1
+    fi
+}
+
+function set_variable () {
+
+    local var=$1
+    local set=$2
+
+    _debug "Setting env variable: $var=$2"
+    export $var="$2"
+}
+
+function launch_impi () {
+    if [ $MODE == "native" ];then
+        LAUNCHER="mpiexec.hydra"
+        verify_launcher_avail $LAUNCHER
+
+	if [ ${RM} == "slurm" ];then
+            cmd="$LAUNCHER -bootstrap slurm $@"
+	elif [ ${RM} == "pbspro" ];then
+	    cmd="$LAUNCHER -rmk pbs $@"
+	fi
+
+	_info "Launch cmd = $cmd (family=$LMOD_FAMILY_MPI)"
+        $cmd
+    fi
+}
+
+function launch_mpich () {
+
+    if [[ ${RM} == "slurm" ]];then
+	if [[ ${OHPC_MPI_LAUNCHERS} =~ "pmix" ]];then
+            verify_launcher_avail srun
+            cmd="srun --mpi=pmix --cpu-bind=cores $@"
+	else
+	    verify_launcher_avail mpiexec.hydra
+	    cmd="mpiexec.hydra -bind-to core -bootstrap slurm $@"
```

Comment:
I understand why you went the route to add some additional command-line options here, but I definitely would not like to maintain two prun variants.  Ideally, it would be nice to just set additional env vars in the test script(s) to get desired binding. I did look at MPICH and see that they have a `HYDRA_BINDING` var which is supposed to accommodate this, but it doesn't look to work in the latest stable release. I did find a patch that will query this var and will include it in the next mpich build for v1.3.5 (https://github.com/openhpc/ohpc/issues/750).  

I think you can do most things in openmpi with env vars as well, but I didn't try this exact case. So, we may not need any mods to prun at all.  If we can't do everything with env vars, I would suggest that as opposed to having a hacked prun separately, that we expand prun to allow you to append additional desired command line arguments and we can commit to that for the 1.3.6 branch if it's necessary.

Long story short, I would suggest you drop `geopm-run` and assume that (1) you can get core binding with env vars or (2) we'll extend the existing prun to allow you add the desired command-line options.  If you are ok with that route and update the PR to remove geopm-prun, I'll land it on the v1.3.6 branch.

Reply:
Chris and I have been working up a solution today that does not involve forking prun.  I've tested it and it's working as well as the solution currently in this PR.  I'll update the PR with that work in a few minutes.

For reference, the patch is here: c0a383802cdeea066536e7f5276fa1f03d3d5fe3.

Reply:
Ok, the latest update removes geopm-prun and adds the necessary command line args for core binding through export bash function overrides for mpiexec.hydra, mpirun, and srun.  Let me know what you think on this approach.

Reply:
I see this as an improvement but still not crazy about it since it will be sensitive to job launch changes in the future, and that's the motivation for having something like `prun` in the first place. I'll land this on 1.3.6 and we can take the time to fix MPICH and prun if necessary to make the testing process clean and hopefully semi-future proof.  

Thanks for the work on this.